index.js

```diff
@@ -100,7 +117,7 @@ deepmerge.all = function deepmergeAll(array, options) {
 
 	return array.reduce(function(prev, next) {
 		return deepmerge(prev, next, options)
-	}, {})
+	}, getDestinationObject(array, undefined, options))
```

Comment:
Following on from my other comment, this is how `getDestinationObject` would be called from `deepmergeAll`:

```js
const target = array[0];
const destination = getDestinationObject(target, options)
const source = destination === target ? array.slice(1) : array

return source.reduce(function(prev, next) {
	return deepmerge(prev, next, options)
}, destination)
```